darts/tests/utils/historical_forecasts/test_historical_forecasts.py

```diff
@@ -2630,6 +2648,342 @@ def test_probabilistic_optimized_hist_fc_regression(self, config):
+            **ts,
+            **hf_args,
+            data_transformers={
+                key_: Pipeline([val_]) for key_, val_ in hf_scaler.items()
+            },
+        )
+
+        # verify that the results are identical when using single Scaler or a Pipeline
+        assert len(hf_auto) == len(hf_auto_pipeline) == 2
+        self.helper_compare_hf(hf_auto, hf_auto_pipeline, last_points_only)
+
+        # optimized historical forecast since horizon_length <= ocl and retrain=False
+        if not retrain:
+            opti_hf_args = {**hf_args, **{"enable_optimization": True}}
+            assert opti_hf_args["enable_optimization"]
+
+            opti_hf_auto = model.historical_forecasts(
+                **ts,
+                **opti_hf_args,
+                data_transformers=hf_scaler,
+            )
+            assert len(opti_hf_auto) == len(hf_auto) == 2
+            self.helper_compare_hf(hf_auto, opti_hf_auto, last_points_only)
+
+        # for 2nd to last historical forecast
+        manual_hf_0 = self.helper_manual_scaling_prediction(
+            model, ts, hf_scaler, retrain, -ocl - 1, ocl
+        )
+        # for last historical forecast
+        manual_hf_1 = self.helper_manual_scaling_prediction(
+            model, ts, hf_scaler, retrain, -ocl, ocl
+        )
+
+        # verify that automatic and manual pre-scaling produce identical forecasts
+        if last_points_only:
+            tmp_ts = TimeSeries.from_times_and_values(
+                times=manual_hf_1.time_index[-2:],
+                values=np.array([manual_hf_0.values()[-1], manual_hf_1.values()[-1]]),
+                columns=manual_hf_0.components,
+            )
+            self.helper_compare_hf(tmp_ts, hf_auto, last_points_only)
+        else:
+            self.helper_compare_hf(
+                hf_auto, [manual_hf_0, manual_hf_1], last_points_only
+            )
+
+    def test_historical_forecasts_with_scaler_errors(self):
+        """"""
+        ocl = 2
+        model = LinearRegressionModel(lags=5, output_chunk_length=ocl)
+        model.fit(self.sine_univariate1)
+
+        hf_args = {
+            "start": -ocl - 1,
+            "start_format": "position",
+            "forecast_horizon": ocl,
+            "verbose": False,
+        }
+
+        # retrain=False and unfitted data transformers
+        with pytest.raises(ValueError) as err:
+            _ = model.historical_forecasts(
+                **hf_args,
+                series=self.sine_univariate1,
+                data_transformers={"series": Scaler()},
+                retrain=False,
+            )
+        assert str(err.value).startswith(
+            "All the fittable entries in `data_transformers` must already be fitted when `retrain=False`."
+        )
+
+        # retrain=False and data transformers fitted on multplie series with fit_global=False
+        with pytest.raises(ValueError) as err:
+            tmp_scaler = Scaler(global_fit=False)
```

Comment:
we need to add support for this (see my previous comment in `darts/utils/historical_forecasts/utils.py`)